python/python/bystro/imputation/soft_impute_cv.py

```diff
@@ -0,0 +1,144 @@
+from typing import Any, Dict
+import numpy as np
+
+from bystro.imputation.fancyimpute.soft_impute import SoftImpute
+
+
+def nan_with_probability(X, p, rng):
```

Comment:
Do we want to copy X within this method to avoid mutation at the function level rather than requiring the user to X.copy()?

Else, let's document that this mutates the input X.

Reply:
Oh we definitely want to change that, didn't realize it was mutated.